### PR TITLE
Interaction Menu - fix a crash when loading savegames

### PR DESCRIPTION
**Pull request by commy2**
**When merged this pull request will:**
- Currently any mission with sufficiently enough soldiers placed crashes when loading a savegame, because the game runs out of memory when deserializing arrays.
- The addInteractionToClass function from interact_menu with allowInheritance flag set adds an (retroactively applied) Init event handler to the object class for every added action, so the actions can be added to descendant classes of the base class the action was added to.
- ACE Medical adds a lot of actions to the CAManBase base class.
- This means ~250 more Init events on CAManBase when Medical is loaded.
- The array is referenced on every CAManBase object and thus creates a big save file, which cannot be loaded without running out of memory.

- This PR rewrites the allowInheritance part of the addInteractionToClass function to drastically reduce the memory usage.
- Realistically this is just a BI bug we will keep running into, but we can avoid it with this memory optimization.

### CODE REVIEW

**Comment by commy2** [comment]
Thanks @Dystopian :)

**Comment by PabstMirror** [comment]
Most inherited actions were for CAManBase (from medical)
So I added special handling for just that class,
it lets us do one check of `isKindOf && pushBackUnique` per unit init instead of once for each action

117 Units
Before
Init: 0.871338s
Place 8 Units: 0.0100098s / 8 = 1.25122ms

After
Init: 0.585205s
Place 8 Units: 0.000976563s / 8 = 0.12207ms

**Comment by PabstMirror** [comment]
This will break WIP https://github.com/acemod/ACE3/pull/6295
But I think it will actually be much easier to implement

File: addons/interact_menu/XEH_preInit.sqf [hunk 1]
```diff
@@ -75,14 +75,20 @@ GVAR(lastTimeSearchedActions) = -1000;
...
-};
+GVAR(inheritedActions) = [];
+GVAR(inheritedManActions) = []; // Actions only for CAManBase
+GVAR(inheritedManClasses) = [];
 
 ["All", "InitPost", {
     params ["_object"];
     private _type = typeOf _object;
 
+    if ((_object isKindOf "CaManBase") && {GVAR(inheritedManClasses) pushBackUnique _type != -1}) then {
```
Comment (commy2) [inline addons/interact_menu/XEH_preInit.sqf]:
```suggestion
  if ((_object isKindOf "CAManBase") && {GVAR(inheritedManClasses) pushBackUnique _type != -1}) then {
```

File: addons/interact_menu/functions/fnc_addActionToClass.sqf [hunk 2]
```diff
@@ -28,21 +28,24 @@ if (!params [["_objectType", "", [""]], ["_typeNum", 0, [0]], ["_parentPath", []
 TRACE_4("params",_objectType,_typeNum,_parentPath,_action);
 
 if (param [4, false, [false]]) exitwith {
-    if (isNil QGVAR(inheritedActions)) then {
-        GVAR(inheritedActions) = [];
-    };
-
-    private _addedClasses = [];
-    GVAR(inheritedActions) pushBack [_addedClasses, _objectType, _typeNum, _parentPath, _action];
+    if (_objectType == "CaManBase") then {
```
Comment (commy2) [inline addons/interact_menu/functions/fnc_addActionToClass.sqf]:
```suggestion
    if (_objectType == "CAManBase") then {
```

**Pull request merged**